django/db/backends/base/creation.py

```diff
@@ -59,6 +59,8 @@ def create_test_db(self, verbosity=1, autoclobber=False, serialize=True, keepdb=
         settings.DATABASES[self.connection.alias]["NAME"] = test_database_name
         self.connection.settings_dict["NAME"] = test_database_name
 
+        self.connection._run_in_test_case = True
```

Comment:
If we keep this approach, it's likely to be adopted elsewhere so we should choose the name carefully. That said, I'm relatively ambivalent about what it should be -- any opinions?